pkg/diagnose/runner.go

```diff
@@ -376,54 +379,48 @@ func RunInAgentProcess(diagCfg diagnosis.Config, deps SuitesDepsInAgentProcess)
 	return getDiagnosesFromCurrentProcess(diagCfg, suites)
 }
 
-// RunStdOutInAgentProcess enumerates registered Diagnose suites and get their diagnoses
-// for human consumption
-func RunStdOutInAgentProcess(w io.Writer, diagCfg diagnosis.Config, deps SuitesDepsInAgentProcess) error {
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		return RunInAgentProcess(diagCfg, deps)
-	})
+// RunLocalCheck runs locally the checks created by the registries.
+func RunLocalCheck(diagCfg diagnosis.Config, registries ...func(*diagnosis.Catalog)) ([]diagnosis.Diagnoses, error) {
+	suites := buildCustomSuites(registries...)
+	return getDiagnosesFromCurrentProcess(diagCfg, suites)
 }
 
-// RunStdOutInCLIProcess enumerates registered Diagnose suites and get their diagnoses
-// for human consumption
-func RunStdOutInCLIProcess(w io.Writer, diagCfg diagnosis.Config, deps SuitesDepsInCLIProcess) error {
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		return RunInCLIProcess(diagCfg, deps)
-	})
+// RunDiagnoseStdOut runs the diagnose and outputs the results to the writer.
+func RunDiagnoseStdOut(w io.Writer, diagCfg diagnosis.Config, diagnoses []diagnosis.Diagnoses) error {
+	if diagCfg.JSONOutput {
+		return runStdOutJSON(w, diagnoses)
+	}
+	return runStdOut(w, diagCfg, diagnoses)
 }
 
-// RunStdOutLocalCheck runs locally the checks created by the registries.
-func RunStdOutLocalCheck(w io.Writer, verbose bool, registries ...func(*diagnosis.Catalog)) error {
-	diagCfg := diagnosis.Config{Verbose: verbose, RunLocal: true}
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		suites := buildCustomSuites(registries...)
+func runStdOutJSON(w io.Writer, diagnoses []diagnosis.Diagnoses) error {
 
-		return getDiagnosesFromCurrentProcess(diagCfg, suites)
-	})
-}
+	var count counters
 
-func runStdOut(w io.Writer, diagCfg diagnosis.Config, run func(diagnosis.Config) ([]diagnosis.Diagnoses, error)) error {
-	if w != color.Output {
-		color.NoColor = true
+	for _, diag := range diagnoses {
+		diagJSON, err := json.MarshalIndent(diag, "", "  ")
+		if err != nil {
+			body, _ := json.Marshal(map[string]string{"error": fmt.Sprintf("marshalling diagnose results to JSON: %s", err)})
+			fmt.Fprintln(w, string(body))
+			return err
+		}
+		fmt.Fprintln(w, string(diagJSON))
+		for _, d := range diag.SuiteDiagnoses {
+			count.increment(d.Result)
+		}
```

Comment:
Suggestion: Why don't we move the counters struct as part of the the `Diagnosis`? 

We would have to update the value when collecting the diagnose on `RunDiagnose`. That would allow use to avoid duplicating the logic in `runStdOutJSON` and `runStdOut`, and even better we could get rid the `summary` function and use for the JSON version the json tags for the structs and for the text version, we would just iterate over the values and print them. 

What do you think?

Reply:
Totally agree, I thought of this too but last time I touch to these structs people said to me to avoid to do it. however from my pov I strongly approve to integrate it

Reply:
done, I can remove summary but it will need to move the logic of this function to runStdout idk if this is worth, wdyt ?